modules/mobile/spec/request/prescriptions_request_spec.rb

```diff
@@ -0,0 +1,255 @@
+                                                   'code' => '104',
+                                                   'status' => '400' }] })
+        end
+      end
+    end
+
+    describe 'sorting parameters' do
+      context 'sorts by ASC refill status' do
+        let(:params) { { sort: 'refill_status' } }
+
+        it 'sorts prescriptions by ASC refill_status', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get '/mobile/v0/health/rx/prescriptions', params: params, headers: iam_headers
+          end
+
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+          expect(response.parsed_body['data'].map { |d| d.dig('attributes', 'refillStatus') }).to eq(
+            %w[active
+               discontinued
+               discontinued
+               expired
+               expired
+               hold
+               refillinprocess
+               refillinprocess
+               refillinprocess
+               refillinprocess]
+          )
+        end
+      end
+
+      context 'sorts by DESC refill status' do
+        let(:params) { { sort: '-refill_status' } }
+
+        it 'sorts prescriptions by DESC refill_status', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get '/mobile/v0/health/rx/prescriptions', params: params, headers: iam_headers
+          end
+
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+          expect(response.parsed_body['data'].map do |d|
+                   d.dig('attributes',
+                         'refillStatus')
+                 end).to eq(%w[submitted refillinprocess refillinprocess refillinprocess refillinprocess refillinprocess
+                               refillinprocess refillinprocess hold expired])
+        end
+      end
+
+      context 'invalid sort option' do
+        let(:params) { { sort: 'quantity' } }
+
+        it 'sorts prescriptions by refill_status', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get '/mobile/v0/health/rx/prescriptions', params: params, headers: iam_headers
+          end
+
+          expect(response).to have_http_status(:bad_request)
+          expect(response.parsed_body).to eq({ 'errors' =>
+                                                [{ 'title' => 'Invalid sort criteria',
+                                                   'detail' =>
+                                                    '"quantity" is not a valid sort criteria for "Prescription"',
+                                                   'code' => '106',
+                                                   'status' => '400' }] })
+        end
+      end
+    end
+
+    describe 'all parameters' do
```

Comment:
IMO, this is redundant. The only additional value this brings is that it confirms that the 'sort' param is used in pagination link formation, which you could just incorporate in that test if you want to confirm that it works.

Reply:
I see. IMO, you can still test this as part of your pagination tests under the umbrella of "pagination links are formed as expected using provided filters" or whatever. I don't see a need to test pagination links with and without params. The pagination helper has some unit tests, I believe, and I will write more when I refactor it soon.